hack/make-rules/test.sh

```diff
@@ -324,21 +335,19 @@ runTests() {
   # vendor/k8s.io/client-go/1.4/rest: causes cover internal errors
   #                            https://github.com/golang/go/issues/16540
   cover_ignore_dirs="vendor/k8s.io/code-generator/cmd/generator|vendor/k8s.io/client-go/1.4/rest"
-  for path in $(echo ${cover_ignore_dirs} | sed 's/|/ /g'); do
+  for path in ${cover_ignore_dirs//|/ }; do
       echo -e "skipped\tk8s.io/kubernetes/${path}"
   done
 
   printf "%s\n" "${@}" \
     | grep -Ev ${cover_ignore_dirs} \
-    | xargs -I{} -n 1 -P ${KUBE_COVERPROCS} \
+    | xargs -I{} -n 1 -P "${KUBE_COVERPROCS}" \
     bash -c "set -o pipefail; _pkg=\"\$0\"; _pkg_out=\${_pkg//\//_}; \
-      go test ${goflags[@]:+${goflags[@]}} \
-        ${KUBE_RACE} \
-        ${KUBE_TIMEOUT} \
+      go test ${goflags[*]:+${goflags[*]}} \
         -cover -covermode=\"${KUBE_COVERMODE}\" \
         -coverprofile=\"${cover_report_dir}/\${_pkg}/${cover_profile}\" \
         \"\${_pkg}\" \
-        ${testargs[@]:+${testargs[@]}} \
+        ${testargs[*]:+${testargs[*]}} \
```

Comment:
why is this *? this should be @?

Reply:
The following failure was detected after modifying other parts. So, I changed this to "*".
```
In ./hack/make-rules/test.sh line 350:
        ${testargs[@]:+${testargs[@]}} \
        ^----------------------------^ SC2145: Argument mixes string and array. Use * or separate argument.
```

Reply:
a space seperated string is wrong for `args` unless we're doing an eval, which we are not

Reply:
agh this code is hard to read here, so we're running this inside of a large `bash -c` (we should really reconsider that), `*` is fine in that case.

Reply:
This part may not be able to run properly with testargs[@].

my result with `bash -x` is here.

- original code (testargs[@])
```
$ sudo PATH=$PATH KUBE_COVER=y make test-cmd
...
+ xargs '-I{}' -n 1 -P 4 bash -c 'set -o pipefail; _pkg="$0"; _pkg_out=${_pkg//\//_};
                                ^^^ **start quote**
  go test --timeout 600s -cover -covermode="atomic" -coverprofile="/tmp/k8s_coverage/d431ed5f68ae4ddf888439fb96b687a923412204/20190426-152616/${_pkg}/coverage.out"
  "${_pkg}" --kubeadm-path' '/home/user/go/src/k8s.io/kubernetes/cluster/kubeadm.sh       | tee        | grep ".*"' '{}'
                         ^^^ **end quote**
can't load package: package /home/user/go/src/k8s.io/kubernetes/cluster/kubeadm.sh       | tee        | grep ".*": import "/home/user/go/src/k8s.io/kubernetes/cluster/kubeadm.
sh       | tee        | grep \".*\"": cannot import absolute path
+ test_result=123
```

- testargs[*]
```
+ xargs '-I{}' -n 1 -P 4 bash -c 'set -o pipefail; _pkg="$0"; _pkg_out=${_pkg//\//_};
                                ^^^ **start quote**
  go test --timeout 600s -cover -covermode="atomic" -coverprofile="/tmp/k8s_coverage/d431ed5f68ae4ddf888439fb96b687a923412204/20190426-152739/${_pkg}/coverage.out"
  "${_pkg}" --kubeadm-path /home/user/go/src/k8s.io/kubernetes/cluster/kubeadm.sh       | tee        | grep ".*"' '{}'
ok      k8s.io/kubernetes/cmd/kubeadm/test/cmd  20.339s coverage: 45.8% of statements                          ^^^ **end quote**
+ test_result=0
```